draft-ietf-webtrans-http3.md

```diff
@@ -265,6 +265,54 @@ within server-initiated bidirectional streams is WEBTRANSPORT_STREAM.
 TODO: move the paragraph above into a separate draft; define what happens with
 already existing HTTP/3 frames on server-initiated bidirectional streams.
 
+## Resetting Data Streams
+
+A WebTransport endpoint may send a RESET_STREAM or a STOP_SENDING frame for a
+WebTransport data stream.  Those signals are propagated by the WebTransport
+implementation to the application.
+
+A WebTransport application SHALL provide an error code for those operations.
+Since WebTransport shares the error code space with HTTP/3, WebTransport
+application errors for streams are limited to an unsigned 8-bit integer,
+assuming values between 0x00 and 0xff.  WebTransport implementations SHALL
+remap those error codes into an error range where 0x00 corresponds to
+0x52e4a40fa8db, and 0xff corresponds to 0x52e4a40fa9e2.  Note that there are
+code points inside that range of form "0x1f * N + 0x21" that are reserved by
+{{Section 8.1 of HTTP3}}; those have to be accounted for when mapping the error
+codes.  An example pseudocode can be seen in {{fig-remap-errors}}.
+
+~~~~~~~~~~
+    first = 0x52e4a40fa8db
+    last = 0x52e4a40fa9e2
+
+    def webtransport_code_to_http_code(n):
+        return first + n + n // 0x1e
+
+    def http_code_to_webtransport_code(h):
+        assert(first <= h <= last)
+        assert((h - 0x21) % 0x1f != 0)
+        shifted = h - first
+        return shifted - shifted // 0x1f
+~~~~~~~~~~
+{: #fig-remap-errors title="Pseudocode for converting between WebTransport
+application errors and HTTP/3 error codes; here, `//` is integer division" }
+
+WebTransport data streams are associated with sessions through a header at the
+beginning of the stream; resetting a stream may result in that data being
+discarded.  Because of that, WebTransport application error codes are best
+effort, as the WebTransport stack is not always capable of associating the
+reset code with a session.  The only exception is the situation where there is
+only one session on a given HTTP/3 connection, and no intermediaries between
+the client and the server.
```

Comment:
Is the client (or the server) expected to be aware of the presence of intermediaries for this?

Reply:
I think the way I'd put it is that the web application (i.e. the combination of client and server) has to be aware of lack of intermediaries to rely on those.

Reply:
Having the application be aware of intermediaries sounds like a brittle design, as the presence of intermediaries might be out of the control of the application developer